openjdk.test.mauve/build.xml

```diff
@@ -359,18 +359,43 @@ limitations under the License.
 		<delete dir="${openjdk_test_mauve_src_dir}/gnu/testlet/org/omg/PortableServer/POAOperations"/>
 	</target>
 
-	<target name="get-source" depends="download-cvsclient, create-work-dir" unless="openjdk_test_mauve_already_built">
-		<exec dir="${openjdk_test_mauve_work_dir}" executable="java">
-		<arg value="-Dcvs.root=:pserver:anoncvs@sourceware.org:/cvs/mauve"/>
-		<arg value="-jar"/>
-		<arg value="${first_prereqs_root}/cvsclient/org-netbeans-lib-cvsclient.jar"/>
-		<arg value="export"/>
-		<arg value="-r"/>
-		<arg value="HEAD"/>
-		<arg value="mauve"/>
-</exec>
+	<condition property="isJavaHomeSet" value="true">
+	   <isset property="JAVA_HOME"/>
+    </condition>
+
+    <condition property="isTestJDKHomeSet" value="true">
+	  <isset property="TEST_JDK_HOME"/>
+    </condition>
+
+    <target name="get-source-using-java-home" if="isJavaHomeSet"> 
+        <exec dir="${openjdk_test_mauve_work_dir}" executable="java">
+		      <arg value="-Dcvs.root=:pserver:anoncvs@sourceware.org:/cvs/mauve"/>
+		      <arg value="-jar"/>
+		      <arg value="${first_prereqs_root}/cvsclient/org-netbeans-lib-cvsclient.jar"/>
+		      <arg value="export"/>
+		      <arg value="-r"/>
+		      <arg value="HEAD"/>
+		      <arg value="mauve"/>
+	    </exec>
 	</target>
 
+    <target name="get-source-using-test-jdk-home" if="isTestJDKHomeSet"> 
+        <exec dir="${openjdk_test_mauve_work_dir}" executable="TEST_JDK_HOME/bin/java">
+		      <arg value="-Dcvs.root=:pserver:anoncvs@sourceware.org:/cvs/mauve"/>
+		      <arg value="-jar"/>
+		      <arg value="${first_prereqs_root}/cvsclient/org-netbeans-lib-cvsclient.jar"/>
+		      <arg value="export"/>
+		      <arg value="-r"/>
+		      <arg value="HEAD"/>
+		      <arg value="mauve"/>
+	    </exec>
+	</target>
+
+    <target name="get-source" depends="download-cvsclient, create-work-dir" unless="openjdk_test_mauve_already_built">
+        <antcall target="get-source-using-java-home"/>
+        <antcall target="get-source-using-test-jdk-home"/>
+    </target>
```

Comment:
In the case where we have both TEST_JDK_HOME and JAVA_HOME, do we get source twice?

Reply:
I was under the impression that we can only have one of the 2 set at one given time. If there can be a case where both can be set, we'd have to update the logic.

Reply:
We should have both according to TKG https://github.com/adoptium/TKG/blob/07171520f8b0849d922125d35380d45193f9c5bf/makefile#L24.

@SAY-droid427 are you able to reproduce the original issue https://github.com/adoptium/aqa-tests/issues/2387?

Reply:
I was honestly not able to reproduce the original issue using the instructions as mentioned here: https://github.com/adoptium/aqa-tests/issues/2387#issuecomment-828103854
This was the issue I was getting: https://github.com/adoptium/aqa-tests/issues/2387#issuecomment-828116517
Even now I am getting this output:
```
make configure
makefile:105: OPENJDK_SYSTEMTEST_ROOT is /home/sayani/Desktop/aqa-systemtest
makefile:108: *** ERROR: JAVA_HOME not set.  Stop.
```
So I am not sure I am  following the correct steps. A little help in this would really be useful.